bids/grabbids/bids_layout.py

```diff
@@ -63,10 +49,10 @@ def get_metadata(self, path):
     def get_fieldmap(self, path):
         sub = os.path.split(path)[1].split("_")[0].split("sub-")[1]
         fieldmap_set = {}
-        for file in self.get(subject=sub,
-                             type='(phase1|phase2|phasediff|epi|fieldmap)',
+        type_ = '(phase1|phase2|phasediff|epi|fieldmap)'
+        for file in self.get(subject=sub, type=type_,
                              extensions=['nii.gz', 'nii']):
-            metadata = self.get_metadata(file.filename)
+            metadata = self.get_metadata(file.filename, type=type_)
```

Comment:
`get_metadata` needs to be always limited to `type` of the file that is being used to query (making the `type` argument redundant). Couldn't you get the `type` from the input filename in `get_metadata` and pass it to `get_nearest`?